### ISSUE TITLE
The allow_wraparound flag in lib/Crypto/Util/Counter.py is non-sense

### ISSUE THREAD

**Issue by cryptopathe** [pycrypto/pycrypto#104]
Dear pyCrypto developpers, 

I can't figure out any useful situation where the `allow_wraparound` flag defined in `lib/Crypto/Util/Counter.py` would be useful in practice. Instead, allowing the counter to wrap in the CTR mode can open a serious security issue (one can XOR the portions of the ciphertext generated with the same counter values, and one immediately gets the XOR of the corresponding parts of the plaintexts, which is a serious problem for non-random plaintexts).

If you split the CTR initial value into a nonce and a random initial counter value, and that you get exceptions because of the counter wrapping around, then it is because your counter has probably a too small bit width for your application. With a proper counter size, this situation should only happen with a probability that is negligible in practice. 

I have quickly searched on Github examples where allow_wraparound would be set to True, and found none. 

In summary, this option, introduced in 2.1.0alpha2, can IMHO only help developers to write bad crypto code, and it should be removed.

**Comment by dlitz** [comment]
I think you're probably right.

When I designed `Counter` 7 years ago, I was probably worried about a split nonce being too small and I imagined that people might deal with this by just using a full 128-bit random nonce and relying on their dispersion to avoid block collisions.  Another reason it might have been useful was to help avoid a timing side-channel in code where authentication happens after decryption.

At this point, though, I don't know of any protocol that actually does this, and the world has moved on to better things like GCM, SIV, and chacha20poly1305.  Meanwhile, I *have* seen confusion about the `allow_wraparound` option.

Thanks!
